### PR TITLE
attract-mode: 2.6.2 -> 2.7.0-unstable-2024-08-02

### PR DESCRIPTION
**Pull request by emilazy**
## Description of changes

Includes fixes for FFmpeg 7.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/at/attract-mode/package.nix [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  src = fetchFromGitHub {
+    owner = "mickelson";
+    repo = "attract";
+    rev = "6ab07d2e23a447b8cf56e927a3fe707322820024";
+    hash = "sha256-1RmwNebFUvnbUtPNxTAFlnZK/2DCmJWAYZtP1y9xepU=";
+  };
+
+  patches = [
+    # Build Objective-C++ files with the C++ compiler
+    # https://github.com/mickelson/attract/pull/757
```
Comment (philiptaron) [inline pkgs/by-name/at/attract-mode/package.nix:33]: Now merged; can drop patch and just update to that revision.

**Review by philiptaron — CHANGES_REQUESTED** [review]
LGTM % the update to now-merged PR.

File: pkgs/by-name/at/attract-mode/package.nix (hunk 1 above)
Comment (emilazy) [reply pkgs/by-name/at/attract-mode/package.nix]: Totally missed that getting merged! Pushed again, thanks.

**Comment by philiptaron** [comment]
tidyf caught the dead code 👁️ 🗿 👁️!

File: pkgs/by-name/at/attract-mode/package.nix [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+stdenv.mkDerivation {
+  pname = "attract-mode";
+  version = "2.7.0-unstable-2024-08-02";
+
+  src = fetchFromGitHub {
+    owner = "mickelson";
+    repo = "attract";
+    rev = "6ed3a1e32a519608c0b495295cc4c18ceea6b461";
+    hash = "";
```
Comment (philiptaron) [inline pkgs/by-name/at/attract-mode/package.nix:28]: empty hash 👀

Comment (emilazy) [reply pkgs/by-name/at/attract-mode/package.nix]: Uhh sorry I feel like I’ve been at the bottom of an abandoned mineshaft for a year after https://github.com/NixOS/nixpkgs/pull/332035 and apparently I’m frazzled 🫠

Fixed!

**Review by philiptaron — APPROVED** [review]
Thanks!

**Pull request merged**